src/controller.c

```diff
@@ -84,6 +116,53 @@ void boss_mode() {
     clear();
 }
 
+void pause_mode() {
+    mvprintw(20, GRID_W+5, "*** PAUSED ***");
+
+    // wait until game resumed
+    while (getch() != RESUME_KEY);
+    clear();
+}
+
+void quit_mode() {
+    const char* menu_title = "*** QUIT? ***";
+    const char* menu_items[] = { "yes", "no" };
+    const int num_menu_items = 2;
+    int curr_selection = 0;
+
+    // Until the user presses the select key
+    char user_input;
+    while (user_input != SELECT_KEY) {
+        // Non-blocking user-input
+        timeout(10);
+        user_input = getch();
+
+        // Navigate through the menu based on user-input
+        switch (user_input) {
+            case UP_KEY:
+                decrement_with_min(&curr_selection, 0);
+                break;
+            case DOWN_KEY:
+                increment_with_max(&curr_selection, num_menu_items-1);
+                break;
+        }
+
+        // Render the menu
+        render_menu(menu_title, menu_items, num_menu_items, curr_selection,
+                GRID_W+5, 20);
+    }
+
+    // Take action based on what the user selected from the menu
+    switch (curr_selection) {
+        case 0: // yes, quit
+            kill(getpid(), SIGINT);
```

Comment:
Cleaner way would be to pass in state->mode and change it to SHUTDOWN. No error message output